### PR TITLE
feat(elements): Add SafeIdentifier & Salutation [SDK-1501]

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

Adds a basic Sign In `SafeIdentifier` and `Salutation` component.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/internals/machines/sign-in/types/router.types.ts [hunk 1]
```diff
@@ -93,3 +93,18 @@ export interface SignInRouterSchema {
...
+export type SignInStateValue = any; // TODO: Update
+
+export type SignInRouterSnapshot = MachineSnapshot<
+  SignInRouterContext,
+  SignInRouterEvents,
+  SignInChildren,
+  SignInStateValue,
+  SignInRouterTags,
+  SignInOuptut
+>;
```
Comment (BRKalow) [inline packages/elements/src/internals/machines/sign-in/types/router.types.ts:110]: What's this change for? 👀

File: packages/elements/src/react/sign-in/index.ts [hunk 2]
```diff
@@ -14,6 +14,12 @@ export {
   SignInVerification as Verification,
 } from './verifications';
 
+export {
+  SignInName as Name,
```
Comment (BRKalow) [inline packages/elements/src/react/sign-in/index.ts:18]: Do we need `Name`? Is it used somewhere in our existing flow?

File: packages/elements/src/react/sign-in/identifiers.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  return SignInRouterCtx.useSelector(SignInSafeIdentifierSelector);
+}
+
+export function SignInSalutation(): string {
+  return SignInRouterCtx.useSelector(SignInSalutationSelector);
+}
+
+export function SignInName(): string {
+  return SignInRouterCtx.useSelector(SignInNameSelector);
+}
```
Comment (BRKalow) [inline packages/elements/src/react/sign-in/identifiers.tsx:19]: I'm wondering if we wrap these in a `<span>` to ensure we're consistently rendering markup from our components. The alternative if someone wants to style these now is to wrap them in an element, which I would say isn't consistent with the rest of the component surface at the moment (I think we're always rendering markup for components that render something).

File: packages/elements/src/internals/machines/utils/formatters.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  firstName,
+  lastName,
+  identifier,
+}: {
+  firstName: string | undefined;
+  lastName: string | undefined;
+  identifier: string | undefined;
+}): string {
+  return (firstName && formatName(firstName)) || (lastName && formatName(lastName)) || identifier || '';
+}
```
Comment (BRKalow) [inline packages/elements/src/internals/machines/utils/formatters.ts:24]: 💭 this is potentially something that we'd want to get from FAPI instead.

File: packages/elements/src/internals/machines/sign-in/types/router.types.ts (hunk 1 above)
Comment (tmilewski) [reply packages/elements/src/internals/machines/sign-in/types/router.types.ts:110]: A central place to retain the type for a SignInRouter snapshot... as used largely by `useSelector`.

File: packages/elements/src/react/sign-in/index.ts (hunk 2 above)
Comment (tmilewski) [reply packages/elements/src/react/sign-in/index.ts:18]: Felt like a nicety given it's use in `Salutation`, such that it doesn't fall back to a phone number or email.

Comment (BRKalow) [reply packages/elements/src/react/sign-in/index.ts:18]: I'd say let's remove for now unless we get feedback asking for it specifically. My main signal is that we're not using it in the fixture app.

File: packages/elements/src/react/sign-in/identifiers.tsx (hunk 3 above)
Comment (tmilewski) [reply packages/elements/src/react/sign-in/identifiers.tsx:19]: Not always (ex: `SignIn`). That said, I'm torn. 

On one hand, I agree. 
On another, is that really a problem? 

Meaning do we want to surface more code and force them to have additionaly elements around something they'll likely wrap in an alternate way (similarly to how I've used them in the example sign-in page)?

File: packages/elements/src/internals/machines/utils/formatters.ts (hunk 4 above)
Comment (tmilewski) [reply packages/elements/src/internals/machines/utils/formatters.ts:24]: Definitely, though in the mean time... 🤷

Comment (BRKalow) [reply packages/elements/src/internals/machines/utils/formatters.ts:24]: Maybe we can leave a comment as a breadcrumb? 😄 e.g.

```
// TODO: ideally the derivation of these values lives in FAPI and comes back directly from the API
```

**Comment by tmilewski** [comment]
@BRKalow Updated

File: packages/elements/src/react/sign-in/identifiers.tsx (hunk 3 above)
Comment (BRKalow) [reply packages/elements/src/react/sign-in/identifiers.tsx]: It's not a problem really, just more of a consistency thing and being predictable in how our components behave. We can always adjust down the road though, so happy to ship with just text here.

File: packages/elements/examples/nextjs/app/sign-in/[[...sign-in]]/page.tsx [hunk 5]
```diff
@@ -188,7 +210,9 @@ export default function SignInPage() {
             <Verification name='reset_password_email_code'>
               <H3>Verify your email</H3>
 
-              <P>Please check your email for a verification code...</P>
+              <P className='text-sm'>
+                Please check your <SafeIdentifier /> for a verification code...
```
Comment (BRKalow) [inline packages/elements/examples/nextjs/app/sign-in/[[...sign-in]]/page.tsx:214]: nit: Should this be something like:
```suggestion
                We've sent a verification code to <SafeIdentifier />.
```

**Review by BRKalow — APPROVED** [review]

**Pull request merged**